doc/user/interop.md

```diff
@@ -89,150 +90,110 @@ Returns true only for the `nil` object.
+Otherwise, if the receiver is a Ruby `Hash`:
 
-`READABLE` will be set if the key is found.
+- `READABLE` will be set if the key is found.
 
-`INSERTABLE` will be set if the hash is not frozen.
+- `INSERTABLE` will be set if the hash is not frozen.
 
-`MODIFIABLE` will be set if the key is found and the hash is not frozen.
+- `MODIFIABLE` and `REMOVABLE`  will be set if the key is found and the hash is
+  not frozen.
 
-`REMOVABLE` will be set if the array is not frozen.
+- `INVOCABLE` and `INTERNAL` will not be set.
 
-Otherwise:
+Otherwise if the name starts with an `@`:
 
-`READABLE` will be set if the object responds to a method of the same name.
+- `READABLE` will be set if the instance variable exists.
 
-`MODIFIABLE` will be set if the object responds to a method of the same name
-appended with `=` and the object is not frozen.
+- `INSERTABLE` will be set if the object is not frozen.
 
-`INSERTABLE` will be false.
+- `MODIFIABLE` and `REMOVABLE` will be set if the instance variable exists is
+  found and the object is not frozen.
 
-If the key has a leading `@`:
+- `INVOCABLE` will not be set.
 
-`READABLE` will be set if there is an instance variable of that name.
+- `INTERNAL` will be set.
 
-`MODIFIABLE` will be set if there is an instance variable of that name and the
-object is not frozen.
+Otherwise:
 
-`INSERTABLE` will be set if the object is not frozen.
+- `READABLE` will be set if the object has a method of that name, or the object
+  has a `[]` method.
 
-`INTERNAL` will be set if there is an instance variable of that name.
+- `INSERTABLE` and `MODIFIABLE` will be set if the object has a `[]=` method.
 
-For all objects and keys:
+- `REMOVABLE` and `INTERNAL` will not be set.
 
-`INVOCABLE` is never set, because currently `KEYS` does not include methods.
+- `INVOCABLE` will be set if the object has a method of that name.
 
-`EXISTING` is set if `READABLE`, `MODIFIABLE`, `INVOCABLE`, `INTERNAL` or
-`REMOVABLE` are set.
+For all objects:
 
-`REMOVABLE` is always set to false.
+- `EXISTING` is set if `READABLE`, `MODIFIABLE`, `INVOCABLE`, `INTERNAL` or
+  `REMOVABLE` are set.
 
 ### `READ`
 
-The name must be an `int` (small Ruby `Fixnum`), or a Ruby `String` or `Symbol`,
-or a Java `String`.
-
-If the receiver is a Ruby `String` and the name is an integer, read a byte from
-the string, ignoring the encoding. If the index is out of range you'll get 0:
-
-`READ(string, integer)` → `string.get_byte(integer)`
-
-Otherwise, if the name starts with `@`, read it as an instance variable:
+If the receiver is a Ruby `Array` or `Hash`, call `receiver[name]`.
 
-`READ(object, "@name")` → `object.instance_variable_get("name")`
+Otherwise if the name starts with an `@` it is read as an instance variable.
 
-Otherwise, if there is a method defined on the object with the same name as
-the name, perform a method call using the name as the called method name:
+Otherwise, if there is a method called `[]` on the receiver, call
```

Comment:
I think "defined `[]`" would be less confusing that "called `[]`" in this context.

Reply:
Fixing.